src/celengine/astrooctree.cpp

```diff
@@ -1,177 +1,121 @@
 
+#include <limits.h>
 #include "astrooctree.h"
 #include "star.h"
 #include "deepskyobj.h"
 
+using namespace std;
 using namespace Eigen;
 
-OctreeNode::OctreeNode(PointType cellCenter, double scale, float starEx, float dsoEx, OctreeNode *parent) :
-    m_children(nullptr),
+OctreeNode::OctreeNode(const Vector3d& cellCenter, double scale, OctreeNode *parent) :
     m_cellCenter(cellCenter),
     m_scale(scale),
-    m_starExclusionFactor(starEx),
-    m_dsoExclusionFactor(dsoEx),
     m_parent(parent)
-{}
+{
+    for(int i = 0; i < 8; i++)
+        m_children[i] = nullptr;
+}
 
 OctreeNode::~OctreeNode()
 {
-    if (m_children != nullptr)
-    {
-        for (auto &i : *m_children)
+    for (auto &i : m_children)
+        if (i != nullptr)
             delete i;
-        delete m_children;
-    }
 }
 
-bool OctreeNode::add(Star *star)
+bool OctreeNode::add(LuminousObject *obj)
 {
-    m_stars.emplace_back(star);
+    m_objects.insert(make_pair(obj->getAbsoluteMagnitude(), obj));
```

Comment:
what will happen if 2/3/etc object have the same magnitude?

Reply:
This is multimap. All should sit here without conflicts.